aspnet/publishing/linuxproduction.rst

```diff
@@ -174,3 +151,83 @@ You can redirect application logs (``STDOUT`` and ``STERR``) in the program sect
 
     tail -f /var/log/hellomvc.out.log
 
+Securing our application
+------------------------
+
+Enable ``AppArmor``
+~~~~~~~~~~~~~~~~~~~
+Linux Security Modules (LSM) is a framework that is part of the Linux kernel since Linux 2.6 that supports different implementations of security modules. ``AppArmor`` is a LSM that implements a Mandatory Access Control system which allows you to confine program to a limited set of resources. Ensure `AppArmor <https://wiki.ubuntu.com/AppArmor>`__ is enabled and properly configured.
+
+Configuring our firewall
+~~~~~~~~~~~~~~~~~~~~~~~~
+Close off all external ports that are not in use. Uncomplicated firewall (ufw) provides a frontend for ``iptables`` by providing a command-line interface for configuring the firewall. Ensure that ``ufw`` is installed and available and configured allow traffic on any ports you may need.
+
+.. code-block:: bash
+
+    sudo apt-get install ufw
+    sudo ufw enable
+
+    sudo ufw allow 80/tcp
+    sudo ufw allow 443/tcp
+
+Securing Nginx
+~~~~~~~~~~~~~~
+
+The default distribution of Nginx isn't configured securely with SSL enabled. To ensure we have all the security features we want, we will build from source.
+
+Download the source and install the build dependencies
+``````````````````````````````````````````````````````
+
+.. code-block:: bash
+
+    # Install the build dependencies
+    sudo apt-get update
+    sudo apt-get install build-essential zlib1g-dev libpcre3-dev libssl-dev libxslt1-dev libxml2-dev libgd2-xpm-dev libgeoip-dev libgoogle-perftools-dev libperl-dev
+
+    # Download nginx 1.10.0 or latest
+    wget http://www.nginx.org/download/nginx-1.10.0.tar.gz
+    tar zxf nginx-1.10.0.tar.gz
+
+Change the Nginx response name
+``````````````````````````````
+
+Edit *src/http/ngx_http_header_filter_module.c*
+
+.. code-block:: c
+
+    static char ngx_http_server_string[] = "Server: Your Web Server" CRLF;
+    static char ngx_http_server_full_string[] = "Server: Your Web Server" CRLF;
+
+
+Configure the options and build
+```````````````````````````````
+
+The PCRE library is required for regular expressions in the location directive for the ngx_http_rewrite_module. The http_ssl_module adds HTTPS protocol support.
```

Comment:
something like
The PCRE library is required for regular expressions. regular expressions are used in the ``location`` directive for the ngx_http_rewrite_module.